src/docs/arch-disagg-layered.dox

```diff
@@ -0,0 +1,107 @@
+- All nodes must use the same URI for the same table
+- All nodes must use byte-wise identical keys, values, and timestamps for the same records.
+- Prepared transactions will behave as described in the design for distributed transactions in
+disaggregated storage
+- Durability is provided by the logical log service, not WiredTiger
+    - Commit durability (using write-ahead log) is not supported for Layered Tables.
+    - Checkpoint durability is asynchronous. The completion of a `WT_SESSION::checkpoint()` call
+    does not guarantee that the resulting checkpoint is on storage and visible to other nodes yet.
+- Table compaction is not supported on layered tables
+- Backup cursors will not return information about layered tables
+- Bulk cursors are not supported on layered tables
+
+@section shared-tables Shared tables
+
+A shared table is a table whose data is stored in a shared storage back-end. A table
+that stores all its data on disk can be called a local or regular table. There is no requirement for
+a stable table to be a shared table, but in all known cases, it is. The ingest table is a local table.
+
+There are also shared tables that are not part of any layered table (e.g., a shared history store
+table or a shared metadata table). The key difference between a regular table and a shared table is
+that the shared one uses the disaggregated block manager.
+
+Generally speaking, a layered table is not a table in the traditional sense, but rather an adapter
+that splits data access between two layers, enabling storage in shared storage.
+
+The following diagram is not meant to strictly represent entities relationships, but rather to aid
+in understanding the high-level design.
+
+\image html shared_table_oop_view.png
+
+@section layered-implementation Implementation aspects
+
+__Note:__ This section is closely tied to the source code and may become outdated if not
+maintained in sync with implementation changes.
+
+@subsection layered-metadata Layered tables metadata representation and related files
+
+Each layered table contains three related entries in metadata:
+
+```
+layered:foo.wt
+    disaggregated=(page_log=palm),ingest="file:foo.wt_ingest",log=(enabled=false),stable="file:foo.wt_stable"
+file: foo.wt_ingest
+file: foo.wt_stable
+```
+
+So the ingest and stable tables can be accessed through those files. The ingest and stable table
+could also have table: prefixed metadata entries if the table was created with
+the `block_manager=disagg,type=layered` config.
+
+Since shared metadata and history store tables are shared but not layered tables, they both have
+`.wt_stable` suffixed metadata entries but no associated `layered:` ones:
+
+```
+file:WiredTigerShared.wt_stable
+file:WiredTigerSharedHS.wt_stable
+```
+
+Despite the presence of `file:`-prefixed metadata entries for both ingest and stable tables, only
+ingest tables have corresponding files on disk in the database directory, since they are local tables.
```

Comment:
As per earlier comment, this is inaccurate. The ingest table is an in-memory table and is node-local.

Reply:
Good catch! I clarified what I've meant there, please check.